backend/src/activitypub/activities/handle.ts

```diff
@@ -287,16 +286,12 @@ export async function handle(
 
 			const notifId = await createNotification(db, 'reblog', targetActor, fromActor, obj)
 
-			await Promise.all([
-				// Add the object in the originating actor's outbox, allowing other
-				// actors on this instance to see the note in their timelines.
-				addObjectInOutbox(db, fromActor, obj, activity.published),
-
-				// Store the reblog for counting
-				insertReblog(db, fromActor, obj),
-
-				sendReblogNotification(db, fromActor, targetActor, notifId, adminEmail, vapidKeys),
```

Comment:
I think I would keep this function call separate. E.g.

```ts
await Promise.all([createReblog(...), sendReblogNotification(...)])
```

Sending a notification is not necessarily part of creating a reblog, hence the fact you had to make it optional.

Reply:
yeah good point 🙂👍